lib/vanagon/driver.rb

```diff
@@ -46,6 +48,24 @@ def load_engine(engine_type, platform, target)
       load_engine_object(engine_type, platform, target)
     end
 
+    # Should we be returning scheduled build hosts instead of 'hardware' engine
+    # hosts?
+    #
+    # REMIND: In the future we will likely allow configurations to specify
+    #   a specific engine. For now, we are introducing this to allow
+    #   transition for just which could use a "scheduled" engine.
+    def scheduled_hosts?(platform)
+      platform.engine && schedule_hardware_allocation?
+    end
+
+    # Should we allow the allocation of hosts via an external scheduler?
+    #
+    # REMIND: This is a feature flag method, triggered by an ENV setting
+    def schedule_hardware_allocation?
+      ENV['VANAGON_SCHEDULED_HARDWARE'] &&
+        !ENV['VANAGON_SCHEDULED_HARDWARE'].empty?
+    end
```

Comment:
The desired engine can be passed to the `build_host_info` command:

```bash
$ bundle exec build_host_info puppet-agent el-7-x86_64 --engine pooler
{"name":"centos-7-x86_64","engine":"pooler"}
```

So you could use that instead of an ENV variable:

```shell
$ bundle exec build_host_info puppet-agent aix-7.1-ppc --engine scheduled
```